src/test/librados/watch_notify.cc

```diff
@@ -90,7 +90,9 @@ void LibRadosWatchNotify::watch_notify2_test_errcb_reconnect(void *arg,
   auto thiz = reinterpret_cast<LibRadosWatchNotify*>(arg);
   ceph_assert(thiz);
   thiz->notify_err = rados_unwatch2(thiz->ioctx, cookie);
-  thiz->notify_cookies.erase(cookie); //delete old cookie
+  if (thiz->notify_cookies.count(cookie)) {
+    thiz->notify_cookies.erase(cookie);
+  }
```

Comment:
> tests: Segmentation fault during handle_error

the title reads like a bug report. it is supposed to be a summary of the change made in the commit.

and i think

```c++
thiz->notify_cookies.erase(cookie);
```

and 

```c++
if (thiz->notify_cookies.count(cookie)) {
     thiz->notify_cookies.erase(cookie);
   }
```

are equivalent. i am not sure what you mean by "notify_cookies set is not initialized yet". if it's just empty, then `thiz->notify_cookies.erase(cookie)` is a noop.

Reply:
Right, just checked it, i thought that's the reason for the seg. Fault 
I'm going to remove that commit, will need to address that if we will hit it again on different test.